staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1365,29 +1364,111 @@ func hasServedCRDVersion(spec *apiextensionsv1.CustomResourceDefinitionSpec, ver
 // buildOpenAPIModelsForApply constructs openapi models from any validation schemas specified in the custom resource,
 // and merges it with the models defined in the static OpenAPI spec.
 // Returns nil models ifthe static spec is nil, or an error is encountered.
-func buildOpenAPIModelsForApply(staticOpenAPISpec *spec.Swagger, crd *apiextensionsv1.CustomResourceDefinition) (proto.Models, error) {
-	if staticOpenAPISpec == nil {
-		return nil, nil
+func buildOpenAPIModelsForApply(staticTypeConverter fieldmanager.TypeConverter, crd *apiextensionsv1.CustomResourceDefinition) (fieldmanager.TypeConverter, error) {
+	// Cannot construct CRD models if native models are not available.
+	// At a minimum need TypeMeta/ObjectMeta since those are injected by builder,
+	// but there may be other native type dependencies found inside $refs.
```

Comment:
Yes, relating to autoscaler and resource.Quantity

Reply:
I had an earlier prototype which did not carry around the type converter, and instead used the zz_generated_openapi definitions. But I changed to this method since the openapi defs may be customized.